### PR TITLE
TS-4683 Adds better error handling on config problems

### PR DESCRIPTION
**Pull request by zwoop**

### CODE REVIEW

**Comment by atsci** [comment]
FreeBSD build *failed*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/460/ for details.

**Comment by atsci** [comment]
Linux build *failed*! See https://ci.trafficserver.apache.org/job/Github-Linux/356/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/461/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/357/ for details.

File: lib/ts/tests.cc [hunk 1]
```diff
@@ -19,7 +19,7 @@
   limitations under the License.
 */
 
-#include "ts/TestBox.h"
+#include "ts/Regression.h"
```
Comment (jpeach) [inline lib/ts/tests.cc]: Is this change related?

**Comment by jpeach** [comment]
👍

File: lib/ts/tests.cc (hunk 1 above)
Comment (zwoop) [inline lib/ts/tests.cc]: Not really, but I noticed it while fixing another problem. I can file a separate Jira if you feel strongly, but I'm certain it's the right fix (without it, we get compiler warnings).

File: plugins/header_rewrite/ruleset.cc [hunk 2]
```diff
@@ -51,7 +51,7 @@ RuleSet::add_condition(Parser &p, const char *filename)
     if (!c->set_hook(_hook)) {
       TSError("[%s] in %s: can't use this condition in hook=%d: %%{%s} with arg: %s", PLUGIN_NAME, filename, _hook,
```
Comment (gtenev) [inline plugins/header_rewrite/ruleset.cc]: @zwoop, do you think it would make sense to print the hook name instead of the number? 
Having the number is a great help for the developers but the operators would not know what it means.

**Comment by zwoop** [comment]
Made the changes per Gancho's suggestions.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/468/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/365/ for details.
